x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/models/list/pipeline_to_list.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { flattenPipelineSection } from './flatten_pipeline_section';
+
+export function pipelineToList(pipeline) {
```

Comment:
When transforming the data structure from `Graph` to `Pipeline`, a static method on the `Pipeline` class is used:

https://github.com/justinkambic/kibana/blob/0a3a7a046660f8fb58fb5a4a8554f353adf903dd/x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/models/pipeline/pipeline.js#L74

WDYT about paralleling that approach here, for consistency? So instead of having a `pipelineToList` function, you'd have a `List` class with properties for `inputs`, `filters`, `outputs`, and `queue` and a static method called `fromPipeline`.

Alternatively you could get rid of the `Pipeline` class and replace it with a `graphToPipeline` function.

Either way, I think it'll be nice if the two conversion mechanisms followed a similar API to make it easier to understand how they fit together.

Reply:
Let's do the first approach. The thought for this had crossed my mind while I was working on this, but I wasn't sure if a new class was necessary since a function can accomplish the same thing. I agree we want consistency, and I think it's easier to understand the code in here if we keep the `Pipeline` class and create a new `List` class. We already have the `Graph` class, maintaining these separate classes will display the polymorphic nature of the information we're conveying. If we just have `Graph` and functions that manipulate it into new structures I think it muddies the waters.